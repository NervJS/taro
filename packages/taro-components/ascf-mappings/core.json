{
  "Button": {
    "size": {
      "ascfProp": "dimension",
      "valueMap": {
        "mini": "XS",
        "small": "S",
        "default": "M",
        "large": "L"
      },
      "required": false,
      "default": "M",
      "docLink": "https://ascf.docs/components/button#dimension"
    },
    "loading": {
      "ascfProp": "isLoading",
      "type": "boolean",
      "required": false,
      "default": false,
      "docLink": "https://ascf.docs/components/button#isloading"
    }
  },
  "Input": {
    "password": {
      "ascfProp": "secretInput",
      "type": "boolean",
      "required": false,
      "default": false,
      "docLink": "https://ascf.docs/components/input#secretinput"
    }
  }
}
