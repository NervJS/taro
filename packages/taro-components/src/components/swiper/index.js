import 'weui'
import Nerv from 'nervjs'
import classNames from 'classnames'
import Swipers from 'swiper'

import 'swiper/dist/css/swiper.min.css'
import './style/index.scss'

let INSTANCE_ID = 0

class SwiperItem extends Nerv.Component {
  render () {
    const cls = classNames('swiper-slide', this.props.className)
    return <div className={cls} style={this.props.style} item-id={this.props.itemId}>{this.props.children}</div>
  }
}

const createEvent = type => {
  let e
  try {
    e = new TouchEvent(type)
  } catch (err) {
    e = document.createEvent('Event')
    e.initEvent(type, true, true)
  }
  return e
}

class Swiper extends Nerv.Component {
  constructor () {
    super(...arguments)
    this.$el = null
    this._id = INSTANCE_ID + 1
    INSTANCE_ID++
    this._$current = 0
  }

  componentDidMount () {
    const {
      autoplay = false,
      interval = 5000,
      duration = 500,
      current = 0,
      displayMultipleItems = 1,
      onChange,
      circular,
      vertical,
      onAnimationfinish,
      spaceBetween
    } = this.props

    const that = this
    const opt = {
      // 指示器
      pagination: { el: `.taro-swiper-${this._id} .swiper-pagination` },
      direction: vertical ? 'vertical' : 'horizontal',
      loop: circular,
      slidesPerView: parseInt(displayMultipleItems, 10),
      initialSlide: parseInt(current, 10),
      speed: parseInt(duration, 10),
      observer: true,
      on: {
        slideChange () {
          let e = createEvent('touchend')
          try {
            Object.defineProperty(e, 'detail', {
              enumerable: true,
              value: {
                current: this.realIndex
              }
            })
          } catch (err) {}
          that._$current = this.realIndex
          onChange && onChange(e)
        },
        transitionEnd () {
          let e = createEvent('touchend')
          try {
            Object.defineProperty(e, 'detail', {
              enumerable: true,
              value: {
                current: this.realIndex
              }
            })
          } catch (err) {}
          onAnimationfinish && onAnimationfinish(e)
        }
      }
    }

    // 自动播放
    if (autoplay) {
      opt.autoplay = {
        delay: parseInt(interval, 10),
        stopOnLastSlide: true,
        disableOnInteraction: false
      }
    }

    // 两端距离
    if (spaceBetween) {
      opt.spaceBetween = spaceBetween
    }

    this.mySwiper = new Swipers(this.$el, opt)
  }

  componentWillReceiveProps (nextProps) {
    if (this.mySwiper) {
      let nextCurrent = 0
      if (nextProps.current === 0) {
        nextCurrent = this._$current || 0
      } else {
        nextCurrent = nextProps.current || this._$current || 0
      }
      // 是否衔接滚动模式
      if (nextProps.circular) {
        if (nextProps.current !== 0) this.mySwiper.slideToLoop(parseInt(nextCurrent, 10)) // 更新下标
      } else {
        if (nextProps.current !== 0) this.mySwiper.slideTo(parseInt(nextCurrent, 10)) // 更新下标
      }

      // 判断是否需要停止或开始自动轮播
      if (this.mySwiper.autoplay.running !== nextProps.autoplay) {
        if (nextProps.autoplay) {
          this.mySwiper.autoplay.start()
        } else {
          this.mySwiper.autoplay.stop()
        }
      }

      this.mySwiper.update() // 更新子元素
    }
  }

  componentWillUnmount () {
    this.$el = null
    if (this.mySwiper) this.mySwiper.destroy()
  }

  render () {
    const { className, style, indicatorColor, indicatorActiveColor } = this.props
    let defaultIndicatorColor = indicatorColor || 'rgba(0, 0, 0, .3)'
    let defaultIndicatorActiveColor = indicatorActiveColor || '#000'
    const cls = classNames(`taro-swiper-${this._id}`, 'swiper-container', className)
    const paginationCls = classNames(
      'swiper-pagination',
      {
        'swiper-pagination-hidden': !this.props.indicatorDots,
        'swiper-pagination-bullets': this.props.indicatorDots
      }
    )
    return (
      <div className={cls} style={style} ref={(el) => { this.$el = el }}>
        <div
          dangerouslySetInnerHTML={{
            __html: `<style type='text/css'>
            .taro-swiper-${this._id} .swiper-pagination-bullet { background: ${defaultIndicatorColor} }
            .taro-swiper-${this._id} .swiper-pagination-bullet-active { background: ${defaultIndicatorActiveColor} }
            </style>`
          }}
        />
        <div className='swiper-wrapper'>{this.props.children}</div>
        <div className={paginationCls} />
      </div>
    )
  }
}

export { Swiper, SwiperItem }
