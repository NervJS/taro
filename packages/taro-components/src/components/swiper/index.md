swiper

## API
| 是否支持 | 属性                    | 类型        | 默认值            | 说明                                                         |
| -------- | ----------------------- | ----------- | ----------------- | ------------------------------------------------------------ |
| √        | indicatorDots           | Boolean     | false             | 是否显示面板指示点                                           |
| √        | indicatorColor          | String      | rgba(0, 0, 0, .3) | 指示点颜色                                                   |
| √        | indicatorActiveColor    | String      | 000               | 当前选中的指示点颜色                                         |
| √        | autoplay                | Boolean     | false             | 是否自动切换                                                 |
| √        | interval                | Number      | 5000              | 自动切换时间间隔                                             |
| √        | duration                | Number      | 500               | 滑动动画时长                                                 |
| √        | current                 | Number      | 0                 | 当前所在滑块的 index                                         |
| √        | bindchange              | EventHandle |                   | current 改变时会触发 change 事件                             |
| √        | circular                | Boolean     | false             | 是否采用衔接滑动                                             |
| ×        | skip-hidden-item-layout | Boolean     | false             | 是否跳过未显示的滑块布局，设为 true 可优化复杂情况下的滑动性能，但会丢失隐藏状态滑块的布局信息 |
| ×        | display-multiple-items  | Number      | 1                 | 同时显示的滑块数量                                           |
| ×        | vertical                | Boolean     | false             | 滑动方向是否为纵向                                           |
| ×        | bindanimationfinish     | EventHandle |                   | 动画结束时会触发 animationfinish 事件                        |
