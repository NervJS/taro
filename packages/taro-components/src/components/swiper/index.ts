export * from './swiper-item'
export * from './swiper'
