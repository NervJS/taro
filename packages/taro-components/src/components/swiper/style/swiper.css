.swiper__container {
  position: relative;
  overflow: hidden;
  min-height: 150px;
  height: 100%;
  width: 100%;
}

.swiper__wrapper {
  position: relative;
  overflow: hidden;
  height: 100%;
}

.swiper__item {
  height: 100%;
  width: 100%;
}

.swiper__pagination {
  position: absolute;
  text-align: center;
  transition: all 0.3s;
  z-index: 10;
}

.swiper__pagination-bullet {
  border-radius: 100%;
  background: rgba(0, 0, 0, 0.3);
  width: 8px;
  height: 8px;
  display: inline-block;
}

.swiper__pagination-bullet.active {
  background: #000000;
}

.swiper__container-horizontal .swiper__pagination {
  bottom: 10px;
  width: 100%;
}

.swiper__container-horizontal .swiper__pagination .swiper__pagination-bullet {
  margin: 0 2.5px;
}

.swiper__container-vertical .swiper__pagination {
  right: 10px;
  top: 50%;
  transform: translate3d(0, -50%, 0);
}

.swiper__container-vertical .swiper__pagination .swiper__pagination-bullet {
  margin: 4px 0;
  display: block;
}
  