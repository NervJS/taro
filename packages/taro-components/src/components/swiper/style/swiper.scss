.swiper {
    &__container {
        position: relative;
        overflow: hidden;
        height: 150px;
        width: 100%;
    }
    &__wrapper {
        position: relative;
        overflow: hidden;
        height: 150px;
    }
    &__item {
        height: 100%;
        width: 100%;
    }
    &__pagination {
        position: absolute;
        text-align: center;
        transition: all 0.3s;
        z-index: 10;
        &-bullet {
            border-radius: 100%;
            background: rgba(0, 0, 0, .3);
            width: 8px;
            height: 8px;
            display: inline-block;
        }
        &-bullet.active {
            background: #000000;
        }
    }
}

.swiper__container-horizontal .swiper__pagination{
    bottom: 10px;
    width: 100%;
    .swiper__pagination-bullet {
        margin: 0 2.5px;
    }
}
.swiper__container-vertical .swiper__pagination{
    right: 10px;
    top: 50%;
    transform: translate3d(0, -50%, 0);
    .swiper__pagination-bullet {
        margin: 4px 0;
        display: block;
    }
}