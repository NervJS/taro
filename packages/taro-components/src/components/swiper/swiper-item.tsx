// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { Component, h, ComponentInterface, Prop, Host } from '@stencil/core'

@Component({
  tag: 'taro-swiper-item-core'
})
export class SwiperItem implements ComponentInterface {
  @Prop() itemId: string

  render () {
    return (
      <Host class='swiper-slide' item-id={this.itemId}/>
    )
  }
}
