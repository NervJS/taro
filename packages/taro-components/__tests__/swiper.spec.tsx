import { h } from '@stencil/core'
import { newSpecPage, SpecPage } from '@stencil/core/testing'

import { Swiper } from '../src/components/swiper/swiper'
import { SwiperItem } from '../src/components/swiper/swiper-item'

describe('Swiper', () => {
  beforeAll(() => {
    // https://github.com/ionic-team/stencil/issues/2056#issuecomment-598086745
    jest.useFakeTimers()
  })

  let page: SpecPage
  const itemStyle = {
    height: '100%'
  }

  it('props', async () => {
    // test props: current, indicator, duration
    // TODO
    const duration = 100
    page = await newSpecPage({
      components: [Swiper, SwiperItem],
      template: () => (<taro-swiper-core duration={duration}>
        <taro-swiper-item-core><div style={itemStyle} /></taro-swiper-item-core>
        <taro-swiper-item-core><div style={itemStyle} /></taro-swiper-item-core>
        <taro-swiper-item-core><div style={itemStyle} /></taro-swiper-item-core>
      </taro-swiper-core>),
    })

    // const swiper = wrapper.find('.swiper-wrapper')
    // const swiperStyles = window.getComputedStyle(swiper)
    // const { width } = swiper.getBoundingClientRect()
    // const indicator = wrapper.find('.swiper-pagination')
    // const bullets = wrapper.findAll('.swiper-pagination-bullet')
    // const bulletAStyle = window.getComputedStyle(bullets[0])
    // const bulletBStyle = window.getComputedStyle(bullets[1])

    // expect(node.duration === duration)
    // expect(swiperStyles.transform === 'matrix(1, 0, 0, 1, 0, 0)')
    // expect(indicator.classList.contains('swiper-pagination-hidden') === true)
    // expect(bulletBStyle.backgroundColor === 'rgba(0, 0, 0, 0.3)')
    // expect(bulletAStyle.backgroundColor === 'rgb(0, 0, 0)')

    // const indicatorColor = 'rgb(153, 153, 153)'
    // const indicatorActiveColor = 'rgb(51, 51, 51)'
    // await wrapper.setProps({
    //   current: 1,
    //   indicatorDots: true,
    //   indicatorColor,
    //   indicatorActiveColor
    // })

    // await delay(duration)

    // expect(swiperStyles.transform === `matrix(1, 0, 0, 1, -${width}, 0)`)
    // expect(indicator.classList.contains('swiper-pagination-hidden') === false)
    // expect(bulletAStyle.backgroundColor === indicatorColor)
    // expect(bulletBStyle.backgroundColor === indicatorActiveColor)

    expect(page.root).toMatchSnapshot()
  })

  it('should autoplay', async () => {
    // TODO
    const interval = 1500
    page = await newSpecPage({
      components: [Swiper, SwiperItem],
      template: () => (<taro-swiper-core autoplay interval={interval}>
        <taro-swiper-item-core><div style={itemStyle} /></taro-swiper-item-core>
        <taro-swiper-item-core><div style={itemStyle} /></taro-swiper-item-core>
        <taro-swiper-item-core><div style={itemStyle} /></taro-swiper-item-core>
      </taro-swiper-core>),
    })

    // const wrapper = await mount(app, scratch)
    // const swiper = wrapper.find('.swiper-wrapper')
    // const swiperStyles = window.getComputedStyle(swiper)
    // const { width } = swiper.getBoundingClientRect()

    // expect(swiperStyles.transform === 'matrix(1, 0, 0, 1, 0, 0)')

    // await delay(interval + wrapper.node.duration)

    // expect(swiperStyles.transform === `matrix(1, 0, 0, 1, -${width}, 0)`)

    expect(page.root).toMatchSnapshot()
  })

  it('should be circular', async () => {
    page = await newSpecPage({
      components: [Swiper, SwiperItem],
      template: () => (<taro-swiper-core circular>
        <taro-swiper-item-core><div /></taro-swiper-item-core>
        <taro-swiper-item-core><div /></taro-swiper-item-core>
        <taro-swiper-item-core><div /></taro-swiper-item-core>
      </taro-swiper-core>),
    })

    // const wrapper = await mount(app, scratch)
    // const swiper = wrapper.find('.swiper-wrapper')
    // expect(swiper.children.length === 5)

    expect(page.root).toMatchSnapshot()
  })

  it('should be vertical', async () => {
    // TODO
    const interval = 1500
    page = await newSpecPage({
      components: [Swiper, SwiperItem],
      template: () => (<taro-swiper-core current={1} vertical autoplay interval={interval}>
        <taro-swiper-item-core><div style={itemStyle} /></taro-swiper-item-core>
        <taro-swiper-item-core><div style={itemStyle} /></taro-swiper-item-core>
        <taro-swiper-item-core><div style={itemStyle} /></taro-swiper-item-core>
      </taro-swiper-core>),
    })

    // const container = wrapper.find('.swiper-container')
    // const swiper = wrapper.find('.swiper-wrapper')
    // const swiperStyles = window.getComputedStyle(swiper)
    // const { height } = swiper.getBoundingClientRect()

    // expect(container.classList.contains('swiper-container-vertical') === true)
    // expect(swiperStyles.transform === `matrix(1, 0, 0, 1, 0, -${height})`)

    // await delay(interval + wrapper.node.duration)

    // expect(swiperStyles.transform === `matrix(1, 0, 0, 1, 0, -${height * 2})`)

    expect(page.root).toMatchSnapshot()
  })

  it('should has previous margin', async () => {
    // TODO
    const previousMargin = 10
    page = await newSpecPage({
      components: [Swiper, SwiperItem],
      template: () => (<taro-swiper-core previousMargin={`${previousMargin}px`}>
        <taro-swiper-item-core><div style={itemStyle} /></taro-swiper-item-core>
        <taro-swiper-item-core><div style={itemStyle} /></taro-swiper-item-core>
        <taro-swiper-item-core><div style={itemStyle} /></taro-swiper-item-core>
      </taro-swiper-core>),
    })

    // const wrapper = await mount(app, scratch)
    // const swiper = wrapper.find('.swiper-container')
    // const swiperStyles = window.getComputedStyle(swiper)

    // expect(swiperStyles.marginLeft === `${previousMargin}px`)

    expect(page.root).toMatchSnapshot()
  })

  it('should display multi items within screen width', async () => {
    // TODO
    page = await newSpecPage({
      components: [Swiper, SwiperItem],
      template: () => (<taro-swiper-core displayMultipleItems={2}>
        <taro-swiper-item-core><div style={itemStyle} /></taro-swiper-item-core>
        <taro-swiper-item-core><div style={itemStyle} /></taro-swiper-item-core>
        <taro-swiper-item-core><div style={itemStyle} /></taro-swiper-item-core>
      </taro-swiper-core>),
    })

    // const swiper = wrapper.find('.swiper-wrapper')
    // const { width } = swiper.getBoundingClientRect()
    // const item = wrapper.find('taro-swiper-item-core')

    // expect(item.style.width === `${width / 2}px`)

    expect(page.root).toMatchSnapshot()
  })

  it('events', async () => {
    // TODO
    const onChange = jest.fn()
    const onAnimationFinish = jest.fn()
    page = await newSpecPage({
      components: [Swiper, SwiperItem],
      template: () => (<taro-swiper-core onChange={onChange} onAnimationFinish={onAnimationFinish}>
        <taro-swiper-item-core><div style={Object.assign({}, itemStyle, { background: 'green' })} /></taro-swiper-item-core>
        <taro-swiper-item-core><div style={Object.assign({}, itemStyle, { background: 'yellow' })} /></taro-swiper-item-core>
        <taro-swiper-item-core><div style={Object.assign({}, itemStyle, { background: 'green' })} /></taro-swiper-item-core>
      </taro-swiper-core>),
    })

    // await wrapper.setProps({
    //   current: 1
    // })

    // await delay(wrapper.node.duration)

    // expect(onChange.calledOnceWith({
    //   current: 1,
    //   source: ''
    // }))

    // expect(onAnimationFinish.calledOnceWith({
    //   current: 1,
    //   source: ''
    // }))

    expect(page.root).toMatchSnapshot()
  })
})
