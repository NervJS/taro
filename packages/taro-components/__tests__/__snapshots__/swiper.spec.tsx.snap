// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swiper events 1`] = `
<taro-swiper-core class="taro-swiper-6" style="overflow: hidden;">
  <!---->
  <div class="swiper-container" style="overflow: visible; margin-right: 0px; margin-left: 0px;">
    <style type="text/css">
      .taro-swiper-6 > .swiper-container > .swiper-pagination > .swiper-pagination-bullet { background: rgba(0, 0, 0, .3) }
              .taro-swiper-6 > .swiper-container > .swiper-pagination > .swiper-pagination-bullet-active { background: #000000 }
    </style>
    <div class="swiper-wrapper">
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%; background: green;"></div>
      </taro-swiper-item-core>
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%; background: yellow;"></div>
      </taro-swiper-item-core>
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%; background: green;"></div>
      </taro-swiper-item-core>
    </div>
    <div class="swiper-pagination swiper-pagination-hidden"></div>
  </div>
</taro-swiper-core>
`;

exports[`Swiper props 1`] = `
<taro-swiper-core class="taro-swiper-0" style="overflow: hidden;">
  <!---->
  <div class="swiper-container" style="overflow: visible; margin-right: 0px; margin-left: 0px;">
    <style type="text/css">
      .taro-swiper-0 > .swiper-container > .swiper-pagination > .swiper-pagination-bullet { background: rgba(0, 0, 0, .3) }
              .taro-swiper-0 > .swiper-container > .swiper-pagination > .swiper-pagination-bullet-active { background: #000000 }
    </style>
    <div class="swiper-wrapper">
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%;"></div>
      </taro-swiper-item-core>
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%;"></div>
      </taro-swiper-item-core>
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%;"></div>
      </taro-swiper-item-core>
    </div>
    <div class="swiper-pagination swiper-pagination-hidden"></div>
  </div>
</taro-swiper-core>
`;

exports[`Swiper should autoplay 1`] = `
<taro-swiper-core class="taro-swiper-1" style="overflow: hidden;">
  <!---->
  <div class="swiper-container" style="overflow: visible; margin-right: 0px; margin-left: 0px;">
    <style type="text/css">
      .taro-swiper-1 > .swiper-container > .swiper-pagination > .swiper-pagination-bullet { background: rgba(0, 0, 0, .3) }
              .taro-swiper-1 > .swiper-container > .swiper-pagination > .swiper-pagination-bullet-active { background: #000000 }
    </style>
    <div class="swiper-wrapper">
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%;"></div>
      </taro-swiper-item-core>
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%;"></div>
      </taro-swiper-item-core>
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%;"></div>
      </taro-swiper-item-core>
    </div>
    <div class="swiper-pagination swiper-pagination-hidden"></div>
  </div>
</taro-swiper-core>
`;

exports[`Swiper should be circular 1`] = `
<taro-swiper-core class="taro-swiper-2" style="overflow: hidden;">
  <!---->
  <div class="swiper-container" style="overflow: visible; margin-right: 0px; margin-left: 0px;">
    <style type="text/css">
      .taro-swiper-2 > .swiper-container > .swiper-pagination > .swiper-pagination-bullet { background: rgba(0, 0, 0, .3) }
              .taro-swiper-2 > .swiper-container > .swiper-pagination > .swiper-pagination-bullet-active { background: #000000 }
    </style>
    <div class="swiper-wrapper">
      <taro-swiper-item-core class="swiper-slide">
        <div></div>
      </taro-swiper-item-core>
      <taro-swiper-item-core class="swiper-slide">
        <div></div>
      </taro-swiper-item-core>
      <taro-swiper-item-core class="swiper-slide">
        <div></div>
      </taro-swiper-item-core>
    </div>
    <div class="swiper-pagination swiper-pagination-hidden"></div>
  </div>
</taro-swiper-core>
`;

exports[`Swiper should be vertical 1`] = `
<taro-swiper-core class="taro-swiper-3" style="overflow: hidden;">
  <!---->
  <div class="swiper-container" style="overflow: visible; margin-top: 0px; margin-bottom: 0px;">
    <style type="text/css">
      .taro-swiper-3 > .swiper-container > .swiper-pagination > .swiper-pagination-bullet { background: rgba(0, 0, 0, .3) }
              .taro-swiper-3 > .swiper-container > .swiper-pagination > .swiper-pagination-bullet-active { background: #000000 }
    </style>
    <div class="swiper-wrapper">
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%;"></div>
      </taro-swiper-item-core>
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%;"></div>
      </taro-swiper-item-core>
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%;"></div>
      </taro-swiper-item-core>
    </div>
    <div class="swiper-pagination swiper-pagination-hidden"></div>
  </div>
</taro-swiper-core>
`;

exports[`Swiper should display multi items within screen width 1`] = `
<taro-swiper-core class="taro-swiper-5" style="overflow: hidden;">
  <!---->
  <div class="swiper-container" style="overflow: visible; margin-right: 0px; margin-left: 0px;">
    <style type="text/css">
      .taro-swiper-5 > .swiper-container > .swiper-pagination > .swiper-pagination-bullet { background: rgba(0, 0, 0, .3) }
              .taro-swiper-5 > .swiper-container > .swiper-pagination > .swiper-pagination-bullet-active { background: #000000 }
    </style>
    <div class="swiper-wrapper">
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%;"></div>
      </taro-swiper-item-core>
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%;"></div>
      </taro-swiper-item-core>
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%;"></div>
      </taro-swiper-item-core>
    </div>
    <div class="swiper-pagination swiper-pagination-hidden"></div>
  </div>
</taro-swiper-core>
`;

exports[`Swiper should has previous margin 1`] = `
<taro-swiper-core class="taro-swiper-4" style="overflow: hidden;">
  <!---->
  <div class="swiper-container" style="overflow: visible; margin-right: 0px; margin-left: 10px;">
    <style type="text/css">
      .taro-swiper-4 > .swiper-container > .swiper-pagination > .swiper-pagination-bullet { background: rgba(0, 0, 0, .3) }
              .taro-swiper-4 > .swiper-container > .swiper-pagination > .swiper-pagination-bullet-active { background: #000000 }
    </style>
    <div class="swiper-wrapper">
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%;"></div>
      </taro-swiper-item-core>
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%;"></div>
      </taro-swiper-item-core>
      <taro-swiper-item-core class="swiper-slide">
        <div style="height: 100%;"></div>
      </taro-swiper-item-core>
    </div>
    <div class="swiper-pagination swiper-pagination-hidden"></div>
  </div>
</taro-swiper-core>
`;
